src/components/graph/layout.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import dagre from 'dagre';
+import { isNode } from 'react-flow-renderer';
+
+const nodeWidth = 150;
+const nodeHeight = 50;
+
+export function getLayoutedElements(elements) {
+    const dagreGraph = new dagre.graphlib.Graph();
+    dagreGraph.setDefaultEdgeLabel(() => ({}));
+    dagreGraph.setGraph({ direction: 'TB', align: 'UL' });
+
+    elements.forEach((el) => {
+        if (isNode(el)) {
+            dagreGraph.setNode(el.id, { width: nodeWidth, height: nodeHeight });
+        } else {
+            dagreGraph.setEdge(el.source, el.target);
+        }
+    });
+
+    dagre.layout(dagreGraph, { debugTiming: true });
+
+    return elements.map((el) => {
+        if (isNode(el)) {
+            const nodeWithPosition = dagreGraph.node(el.id);
```

Comment:
After taking a look at the dagre code, dagreGraph.node simply queries a map of nodes indexed by id, so the execution is quite fast. Doing the opposite way (iterate on all dagreGraph.nodes and searching the node in elements would be less efficient since elements is simply an array)